core/app/models/spree/return_item/exchange_variant_eligibility/same_option_value.rb

```diff
@@ -20,11 +20,17 @@ class SameOptionValue
       # blue pants with 34 waist and 32 inseam
 
       def self.eligible_variants(variant)
-        product_variants = SameProduct.eligible_variants(variant).includes(option_values: :option_type)
-
+        product_variants = SameProduct.eligible_variants(variant)
         relevant_option_values = variant.option_values.select { |ov| option_type_restrictions.include? ov.option_type.name }
+
         if relevant_option_values.present?
-          product_variants.select { |v| (relevant_option_values & v.option_values) == relevant_option_values }
+          variant_ids = Spree::OptionValuesVariant.
+            where(variant_id: product_variants.distinct.pluck(:id)).
+            where(option_value: relevant_option_values).
+            group('variant_id').
+            having('count(1) = ?', relevant_option_values.size).
```

Comment:
`count(1)` is the same as `COUNT(*)`, [yeah](http://stackoverflow.com/questions/181272/what-does-select-count1-from-table-name-on-any-database-tables-mean)? If so can we write it as the latter? :)